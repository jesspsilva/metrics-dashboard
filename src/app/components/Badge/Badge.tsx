import styled from 'styled-components';

const getBadgeColors = (label: string) => {
  switch (label.toLowerCase()) {
    case 'efficiency':
      return {
        backgroundColor: 'var(--light-green)',
        border: 'var(--dark-green)',
        color: 'var(--dark-green)',
      };
    case 'shift':
      return {
        backgroundColor: 'var(--light-blue)',
        border: 'var(--dark-blue)',
        color: 'var(--dark-blue)',
      };
    case 'downtime':
      return {
        backgroundColor: 'var(--light-red)',
        border: 'var(--dark-red)',
        color: 'var(--dark-red)',
      };
    default:
      return {
        backgroundColor: 'var(--light-gray)',
        border: 'var(--dark-gray)',
        color: 'var(--dark-gray)',
      };
  }
};

interface BadgeProps {
  label: string;
}

const BadgeWrapper = styled.span<BadgeProps>`
  padding: 5px 12px;
  border-radius: 5px;
  font-size: 0.9rem;
  text-transform: capitalize;
  color: ${({ label }) => getBadgeColors(label).color};
  background-color: ${({ label }) => getBadgeColors(label).backgroundColor};
  border: 1px solid ${({ label }) => getBadgeColors(label).border};
`;


export default function Badge({label}: BadgeProps) {
  return <BadgeWrapper label={label}>{label}</BadgeWrapper>;
}