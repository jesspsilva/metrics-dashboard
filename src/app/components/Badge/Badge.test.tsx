import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import "jest-styled-components";

import Badge from "./Badge";

describe("Badge", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Badge label="efficiency" />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe("when the label is efficiency", () => {
    it("should render the correct label and styles", async () => {
      const { container } = render(<Badge label="efficiency" />);
      const badge = container.firstChild;
      expect(badge?.textContent).toBe("efficiency");
      expect(badge).toHaveClass(
        "bg-emerald-200 border-emerald-600 text-emerald-800",
      );
    });
  });

  describe("when the label is shift", () => {
    it("should render the correct label and styles", async () => {
      const { container } = render(<Badge label="shift" />);
      const badge = container.firstChild;
      expect(badge?.textContent).toBe("shift");
      expect(badge).toHaveClass("bg-sky-200 border-sky-800 text-sky-800");
    });
  });

  describe("when the label is downtime", () => {
    it("should render the correct label and styles", async () => {
      const { container } = render(<Badge label="downtime" />);
      const badge = container.firstChild;
      expect(badge?.textContent).toBe("downtime");
      expect(badge).toHaveClass("bg-red-200 border-red-800 text-red-800");
    });
  });
});
