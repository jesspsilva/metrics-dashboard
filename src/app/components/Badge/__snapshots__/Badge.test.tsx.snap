// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  padding: 5px 12px;
  border-radius: 5px;
  font-size: 0.9rem;
  text-transform: capitalize;
  color: var(--dark-green);
  background-color: var(--light-green);
  border: 1px solid var(--dark-green);
}

<span
    class="c0"
    label="efficiency"
  >
    efficiency
  </span>
</DocumentFragment>
`;
